### PR TITLE
Add support for lombok.builder.className config property

### LINKED ISSUE #680: lombok.builder.className config which is added in 1.18.10 is not supported
### Short description
Since 1.18.10 Lombok allows configuring default class name for builder classes. See `lombok.builder.className` in https://projectlombok.org/features/Builder but it doesn't seem to be supported in the plugin.

### Version information
 * **IDEA Version:** 2019.2.3
 * **JDK Version:** 1.8.0_212
 * **Lombok Plugin Version:** 0.27-2019.2
 * **Lombok Dependency Version:** 1.18.10

### PR DESCRIPTION
**Pull request by tagae**
Fixes #680

### CODE REVIEW

**Comment by mplushnikov** [comment]
Hello @tagae,
thank you for your work! Your change lets the build fail, because of two failing tests:

> de.plushnikov.intellij.plugin.processor.BuilderTest > testBuilder$BuilderComplex FAILED
>     junit.framework.ComparisonFailure: Types are not equal for element: builder expected:<BuilderComplex.[V]oidBuilder<T>> but was:<BuilderComplex.[v]oidBuilder<T>>
> 

and 

> de.plushnikov.intellij.plugin.processor.BuilderTest > testBuilder$BuilderMethodException FAILED
>     junit.framework.ComparisonFailure: Types are not equal for element: builder expected:<...lderMethodException.[V]oidBuilder> but was:<...lderMethodException.[v]oidBuilder> 

Please add new automatically JUnit test for this functionality. You can find some examples here: https://github.com/mplushnikov/lombok-intellij-plugin/tree/master/src/test/java/de/plushnikov/intellij/plugin/configsystem

Also please add yourself to the list of contributors (in README.md file)

File: src/main/java/de/plushnikov/intellij/plugin/processor/handler/BuilderHandler.java [hunk 1]
```diff
@@ -253,14 +255,23 @@ public String getBuilderClassName(@NotNull PsiClass psiClass, @NotNull PsiAnnota
...
-    return StringUtil.capitalize(rootBuilderClassName + BUILDER_CLASS_NAME);
+
+    return getBuilderClassName(psiClass, relevantReturnType);
+  }
+
+  @NotNull
+  String getBuilderClassName(@NotNull PsiClass psiClass, String returnTypeName) {
+    final ConfigDiscovery configDiscovery = ConfigDiscovery.getInstance();
+    final String builderClassNamePattern = configDiscovery.getStringLombokConfigProperty(BUILDER_CLASS_NAME, psiClass);
+    return StringUtil.replace(builderClassNamePattern, "*", returnTypeName);
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/processor/handler/BuilderHandler.java]: returnTypeName should be capitalized: StringUtil.capitalize(returnTypeName )

**Comment by mplushnikov** [comment]
@tagae do you plan to update your PR?

**Comment by tagae** [comment]
Hi,

Sorry for this late answer. I do plan to update the PR, only that I get time for such work only once every 2 weeks. Tomorrow (Monday) I will fix the tests and remaining changes.

**Comment by tagae** [comment]
Not sure how to interpret the failure from "codecov/patch". Is there anything I should still do?

**Comment by mplushnikov** [comment]
Thank you @tagae ! Just merged your PR.

**Pull request merged**